### PR TITLE
Fix not being able to delete folder in data-model

### PR DESCRIPTION
**Pull request by Nitwel**
Due to folders not being represented as a collection in the DB, the request would fail as it would try to delete a non existing collection.

### CODE REVIEW

**Review by rijkvanzanten — APPROVED** [review]

File: api/src/services/collections.ts [hunk 1]
```diff
@@ -377,7 +377,9 @@ export class CollectionsService {
 		}
 
 		await this.knex.transaction(async (trx) => {
-			await trx.schema.dropTable(collectionKey);
+			if (collectionToBeDeleted!.schema !== null) {
+				await trx.schema.dropTable(collectionKey);
+			}
```
Comment (rijkvanzanten) [inline api/src/services/collections.ts:382]: Lets use the same check as below on 397

```suggestion
			if (collectionToBeDeleted!.schema) {
				await trx.schema.dropTable(collectionKey);
			}
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
@Nitwel thanks! Mind adding a e2e test case for deleting collections? That way we can make sure this doesn't happen again 🙂

**Comment by rijkvanzanten** [comment]
@Nitwel
CleanShot 2022-02-04 at 11 30 45@2x

easy fix here is to use `toEqual` instead of `toBe` 👍🏻

**Review by rijkvanzanten — APPROVED** [review]

**Comment by onordgren** [comment]
Is there any way to force delete a folder pending this to be released? I don't have (easy) direct access to the database so using the app/API would be ideal.

**Comment by onordgren** [comment]
> > Is there any way to force delete a folder pending this to be released? I don't have (easy) direct access to the database so using the app/API would be ideal.
> 
> 
> 
> You should be able to use the folders endpoint to delete it still:
> 
> 
> 
> https://docs.directus.io/reference/system/folders/#delete-a-folder

I tried this last week but got a similar error as in the app. I however solved it by accessing the database directly for now.

**Pull request merged**